sentry-android-okhttp/src/main/java/io/sentry/android/okhttp/SentryOkHttpInterceptor.kt

```diff
@@ -104,6 +123,107 @@ class SentryOkHttpInterceptor(
         }
     }
 
+    private fun captureEvent(request: Request, response: Response) {
+        // not possible to get a parameterized url, but we remove at least the
+        // query string and the fragment.
+        // url example: https://api.github.com/users/getsentry/repos/#fragment?query=query
+        // url will be: https://api.github.com/users/getsentry/repos/
+        // ideally we'd like a parameterized url: https://api.github.com/users/{user}/repos/
+        // but that's not possible
+        var requestUrl = request.url.toString()
+
+        val query = request.url.query
+        if (!query.isNullOrEmpty()) {
+            requestUrl = requestUrl.replace("?$query", "")
+        }
+
+        val urlFragment = request.url.fragment
+        if (!urlFragment.isNullOrEmpty()) {
+            requestUrl = requestUrl.replace("#$urlFragment", "")
+        }
+
+        if (!captureFailedRequests ||
```

Comment:
I think you wanted to move this code to make the check somewhere earlier, right?

Reply:
mm, not sure, should the String contains or Regex take into account the query and fragment? I guess its a trade-off of false positive in both ways.

Reply:
nah, I meant what we could do (and what Alex meant), is that we move up the boolean flag and the status code check, so we do not run unnecessary code when the entire feature is disabled, something like

```kotlin
if (captureFailedRequests && containsStatusCode(response.code) {
  captureEvent(...)
}

// and then inside captureEvent
if (!PropagationTargetsUtils.contain(failedRequestTargets, requestUrl)) {
  return
}
```
This way we won't need to do the `requestUrl` string manipulations at all in case the feature is disabled or the status code is not the failing one.

Reply:
Gotcha, that makes a lot of sense, will do that, thanks.